create_hard_negatives_v2.py

```diff
@@ -884,7 +884,9 @@ def main(
                         continue
                     act = acts_L[j]
                     max_act = max(max_act, act)
-                    token_activations.append(TokenActivationV2.model_construct(s=token, act=act, pos=j))
+                    # only save if act > 0
+                    if act > 0.0:
+                        token_activations.append(TokenActivationV2.model_construct(s=token, act=act, pos=j))
 
```

Comment:
We only save activations that are > 0.0. (We still know the tokens themselves, but don't need to save the values for things that have activation of 0.0)